src/components/notification.js

```diff
@@ -1,132 +1,174 @@
 import React, { Component, Fragment } from 'react'
-import { Link } from 'react-router-dom'
 import { FormattedMessage, defineMessages, injectIntl } from 'react-intl'
+import { connect } from 'react-redux'
+import { Link } from 'react-router-dom'
+
+import { fetchUser } from 'actions/User'
+
+import origin from '../services/origin'
 
-/*
-  This subcomponent turns structured notification objects into dom elements,
-  which wrap an anchor if there is a relevant listing.
-*/
-class HumanReadableNotification extends Component {
+// support other derived notifications in the future
+const NON_PURCHASE_RELATED_MESSAGE = 'A message from Origin that does not involve a listing'
+
+class NotificationMessage extends Component {
```

Comment:
This file seems like it would be cleaner if it just had one class - what would you think about moving this `NotificationMessage` class to its own file?

Reply:
@jordajm done ✅